parasol_cpu/src/proc/assembly.rs

```diff
@@ -155,100 +291,91 @@ macro_rules! define_op {
 define_op! {
     IsaOp,
     DispatchIsaOp,
-    (Register, PtrRegister),
-
-    // Bind input
-    [BindReadOnly (dst dst, 1, PtrRegister) (meta id usize) (meta encrypted bool)],
-
-    // Bind output
-    [BindReadWrite (dst dst, 1, PtrRegister) (meta id usize) (meta encrypted bool)],
+    (Register),
 
     // Load
-    [Load (dst dst, 0, Register) (src src, 1, PtrRegister) (meta width u32)],
+    [0x02 Load (dst dst, 0, Register) (src src, 0, Register) (meta width, 7, u32)],
 
     // Load immediate
-    [LoadI (dst dst, 0, Register) (meta imm u128) (meta width u32)],
+    [0x03 LoadI (dst dst, 0, Register) (meta imm, 44, u128) (meta width, 7, u32)],
```

Comment:
How will large immediates get handled later?

Reply:
There are a few ways one does this on real ISAs:
* Risc-V allows 16-bit immediates. You can use li to load into the lower 16-bits of a 32-bit word and liu to load into the upper 16 bits of a word. So, it takes 2 instructions to load a 32-bit immediate.
* You can load values from the .rodata segment (which we now have)

I'm inclined to just require large constants get loaded from constant memory.